### PR TITLE
[FEATURE] migration and model for CorrelationsScore

### PR DESCRIPTION
**Pull request by stevenzeiler**

### CODE REVIEW

File: api/models/CorrelationScore.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+'use strict';
+module.exports = function(sequelize, DataTypes) {
+  var Validation = sequelize.define('CorrelationScores', {
```
Comment (wilsonianb) [inline api/models/CorrelationScore.js]: `Validation` -> `CorrelationScores`

File: api/models/CorrelationScore.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    },
+    date: {
+      type     : DataTypes.STRING,
+      allowNull: false
+    },
+    quorum: {
+      type     : DataTypes.INTEGER,
+      allowNull: false
+    }
+  }, {
```
Comment (wilsonianb) [inline api/models/CorrelationScore.js]: Add column validations (such as cluster members & date format)

Comment (stevenzeiler) [inline api/models/CorrelationScore.js]: commit 0bc7db5c97671f047fd5bf756fd46ef02879e73a adds validation for cluster members.

Comment (stevenzeiler) [inline api/models/CorrelationScore.js]: date is now being validated as well.

**Comment by stevenzeiler** [comment]
:+1:

**Comment by wilsonianb** [comment]
I cleaned up the commit history and pushed to https://github.com/ripple/validator-registry-api/tree/feature/correlations-table-v2
If that looks good, I can push it to this branch.

File: migrations/20150716112723-create-correlation-scores.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    return queryInterface.createTable('CorrelationScores', {
+      id: {
+        type: Sequelize.INTEGER,
+        primaryKey: true,
+        autoIncrement: true
+      },
+      cluster: {
+        type: Sequelize.ARRAY(Sequelize.STRING),
+        notNull: true
```
Comment (wilsonianb) [inline migrations/20150716112723-create-correlation-scores.js]: Is `notNull` a thing? I've only seen `allowNull`

Comment (stevenzeiler) [inline migrations/20150716112723-create-correlation-scores.js]: The syntax is different for migrations versus models :/

Comment (wilsonianb) [inline migrations/20150716112723-create-correlation-scores.js]: The example here has `allowNull`
http://docs.sequelizejs.com/en/latest/docs/migrations/

Comment (wilsonianb) [inline migrations/20150716112723-create-correlation-scores.js]: FYI: I've been creating models and migrations using sequelize-cli
`sequelize model:create --name Verifications --attributes validation_public_key:string,domain:string,error:string`

File: test/models/CorrelationScore.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import 'sails-test-helper'
+import assert from 'assert'
+import moment from 'moment'
+import _      from 'lodash'
+
+describe('CorrelationScore database table', () => {
+
+  it('should persist a cofficient for validator', done => {
```
Comment (wilsonianb) [inline test/models/CorrelationScore.js]: `cofficient` -> `coefficient`

File: api/models/CorrelationScore.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+'use strict';
+module.exports = (sequelize, DataTypes) => {
+  const CorrelationScores = sequelize.define('CorrelationScores', {
+    cluster: {
+      type     : DataTypes.ARRAY(DataTypes.STRING),
+      allowNull: false,
+      validate: {
+        areValidationPublicKeys: (cluster) => {
+          cluster.forEach(validationPublicKey => {
+            if (validationPublicKey.length !== 52) {
```
Comment (wilsonianb) [inline api/models/CorrelationScore.js]: Let's use this regex:
https://github.com/ripple/validator-registry-api/blob/dfc8b6ab6e2150f010a7ac78f01a483224e84754/api/models/validation.js#L9-L10

File: api/models/CorrelationScore.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    date: {
+      type     : DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        isDate: {
+          msg: 'date must be formatted as YYYY-MM-DD'
+        }
+      }
+    },
+    quorum: {
```
Comment (wilsonianb) [inline api/models/CorrelationScore.js]: Maybe add a validation that the quorum is <= the size of the cluster

File: migrations/20150716112723-create-correlation-scores.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+module.exports = {
+  up: function (queryInterface, Sequelize) {
+
+    return queryInterface.createTable('CorrelationScores', {
+      id: {
```
Comment (wilsonianb) [inline migrations/20150716112723-create-correlation-scores.js]: `id` should also have `allowNull: false`

File: test/models/CorrelationScore.js (hunk 4 above)
Comment (wilsonianb) [inline test/models/CorrelationScore.js]: Also `for validator` -> `for validators`?

File: test/models/CorrelationScore.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+        'n9LigbVAi4UeTtKGHHTXNcpBXwBPdVKVTjbSkLmgJvTn6qKB8Mqz': 0.9911,
+        'n949f75evCHwgyP4fPVgaHqNHxUVN15PsJEZ3B3HnXPcPjcZAoy7': 0.9888833
+      }
+    })
+    .catch(error => {
+      assert.strictEqual(error.message, 'Validation error: only ripple validation public keys allowed')
+      done()
+    })
+  })
+
```
Comment (wilsonianb) [inline test/models/CorrelationScore.js]: Maybe add tests for missing required columns

File: migrations/20150716112723-create-correlation-scores.js (hunk 3 above)
Comment (stevenzeiler) [inline migrations/20150716112723-create-correlation-scores.js]: Thanks good to know, I haven't used the `--attributes` feature yet.

File: migrations/20150716112723-create-correlation-scores.js (hunk 7 above)
Comment (stevenzeiler) [inline migrations/20150716112723-create-correlation-scores.js]: done.

File: api/models/CorrelationScore.js (hunk 5 above)
Comment (stevenzeiler) [inline api/models/CorrelationScore.js]: Nice!

File: test/models/CorrelationScore.js (hunk 8 above)
Comment (stevenzeiler) [inline test/models/CorrelationScore.js]: tests for missing required columns are not in place.

**Comment by wilsonianb** [comment]
LGTM pending optional quorum validation

**Comment by stevenzeiler** [comment]
:100:

File: api/models/CorrelationScore.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+      type     : DataTypes.INTEGER,
+      allowNull: false
+    }
+  }, {
+    validate: {
+      isLessThanOrEqualToClusterSize: function() {
```
Comment (wilsonianb) [inline api/models/CorrelationScore.js]: Maybe add quorum to the name

**Comment by wilsonianb** [comment]
:+1:

**Pull request merged**